server/builds.go

```diff
@@ -27,6 +32,25 @@ func (b *Builds) getInstallationVersion(pr *model.PullRequest) string {
 	return pr.Sha[0:7]
 }
 
+func (b *Builds) dockerRegistryClient(s *Server) (reg *registry.Registry, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = errors.Errorf("panicked while connecting to docker registry: %s", r)
+		}
```

Comment:
This is interesting, does `registry` panic on failed connection often?

Reply:
Yeah, I would also like to know the story here.

Reply:
While testing the first thing I checked was incorrect URL (without http part) and it panicked so I thought of catching that and return the error although I'm checking if the url is of correct format.

Reply:
Hmm, interesting. I would prefer to remove the panic recovery here as it may hide other issues in the future. If we check the URL before using it in the registry, we should avoid this issue entirely anyway.

Thoughts, @jwilander?

Reply:
I'm ok with that 👍

Reply:
Yes, that makes sense. I'll make the change. Thanks